### PR TITLE
Added function to check if the parent object exists in all the resources contained in the provider

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: aci/resource_aci_mgmtinboobzone.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func getRemoteInBManagedNodesZone(client *client.Client, dn string) (*models.InB
 	}
 	mgmtInBZone := models.InBManagedNodesZoneFromContainer(mgmtInBZoneCont)
 	if mgmtInBZone.DistinguishedName == "" {
-		return nil, fmt.Errorf("InBManagedNodesZone %s not found", mgmtInBZone.DistinguishedName)
+		return nil, fmt.Errorf("InB Managed Nodes Zone %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_mgmtinboobzone.go:145]:
```suggestion
		return nil, fmt.Errorf("InBand Managed Nodes Zone %s not found", dn)
```

File: aci/resource_aci_mgmtinboobzone.go [hunk 2]
```diff
@@ -154,7 +154,7 @@ func getRemoteOOBManagedNodesZone(client *client.Client, dn string) (*models.OOB
 	}
 	mgmtOobZone := models.OOBManagedNodesZoneFromContainer(mgmtOobZoneCont)
 	if mgmtOobZone.DistinguishedName == "" {
-		return nil, fmt.Errorf("InBManagedNodesZone %s not found", mgmtOobZone.DistinguishedName)
+		return nil, fmt.Errorf("OoB Managed Nodes Zone %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_mgmtinboobzone.go:157]:
```suggestion
		return nil, fmt.Errorf("OOB Managed Nodes Zone %s not found", dn)
```

File: aci/resource_aci_nodemgmtepg.go [hunk 3]
```diff
@@ -271,7 +271,7 @@ func getRemoteInBandManagementEPg(client *client.Client, dn string) (*models.InB
 	mgmtInB := models.InBandManagementEPgFromContainer(mgmtInBCont)
 
 	if mgmtInB.DistinguishedName == "" {
-		return nil, fmt.Errorf("InBandManagementEPg %s not found", mgmtInB.DistinguishedName)
+		return nil, fmt.Errorf("InBand Management EPg %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_nodemgmtepg.go:274]:
```suggestion
		return nil, fmt.Errorf("InBand Management EPG %s not found", dn)
```

File: aci/resource_aci_nodemgmtepg.go [hunk 4]
```diff
@@ -286,7 +286,7 @@ func getRemoteOutOfBandManagementEPg(client *client.Client, dn string) (*models.
 	mgmtOoB := models.OutOfBandManagementEPgFromContainer(mgmtOoBCont)
 
 	if mgmtOoB.DistinguishedName == "" {
-		return nil, fmt.Errorf("OutOfBandManagementEPg %s not found", mgmtOoB.DistinguishedName)
+		return nil, fmt.Errorf("Out Of Band Management EPg %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_nodemgmtepg.go:289]:
```suggestion
		return nil, fmt.Errorf("Out Of Band Management EPG %s not found", dn)
```

File: aci/resource_aci_pkiexportencryptionkey.go [hunk 5]
```diff
@@ -69,7 +69,7 @@ func getRemoteAESEncryptionPassphraseandKeysforConfigExportImport(client *client
 	}
 	pkiExportEncryptionKey := models.AESEncryptionPassphraseandKeysforConfigExportImportFromContainer(pkiExportEncryptionKeyCont)
 	if pkiExportEncryptionKey.DistinguishedName == "" {
-		return nil, fmt.Errorf("AESEncryptionPassphraseandKeysforConfigExport(andImport) %s not found", pkiExportEncryptionKey.DistinguishedName)
+		return nil, fmt.Errorf("AES Encryption Passphrase and Keys for Config Export(and Import) %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_pkiexportencryptionkey.go:72]:
```suggestion
		return nil, fmt.Errorf("AES Encryption Passphrase and Keys for Config Export and Import %s not found", dn)
```

File: aci/resource_aci_rtctrlattrp.go [hunk 6]
```diff
@@ -247,7 +247,7 @@ func getRemoteRtctrlSetRtMetric(client *client.Client, dn string) (*models.Rtctr
 	}
 	rtctrlSetRtMetric := models.RtctrlSetRtMetricFromContainer(rtctrlSetRtMetricCont)
 	if rtctrlSetRtMetric.DistinguishedName == "" {
-		return nil, fmt.Errorf("rtctrlSetRtMetric %s not found", dn)
+		return nil, fmt.Errorf("rtctrl Set RtMetric %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:250]:
```suggestion
		return nil, fmt.Errorf("Route Control Set Route Metric %s not found", dn)
```

File: aci/resource_aci_rtctrlattrp.go [hunk 7]
```diff
@@ -268,7 +268,7 @@ func getRemoteRtctrlSetRtMetricType(client *client.Client, dn string) (*models.R
 	}
 	rtctrlSetRtMetricType := models.RtctrlSetRtMetricTypeFromContainer(rtctrlSetRtMetricTypeCont)
 	if rtctrlSetRtMetricType.DistinguishedName == "" {
-		return nil, fmt.Errorf("rtctrlSetRtMetricType %s not found", dn)
+		return nil, fmt.Errorf("rtctrl Set RtMetric Type %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:271]:
```suggestion
		return nil, fmt.Errorf("Route Control Set Route Metric Type %s not found", dn)
```

File: aci/resource_aci_rtctrlattrp.go [hunk 8]
```diff
@@ -289,7 +289,7 @@ func getRemoteRtctrlSetNh(client *client.Client, dn string) (*models.RtctrlSetNh
 	}
 	rtctrlSetNh := models.RtctrlSetNhFromContainer(rtctrlSetNhCont)
 	if rtctrlSetNh.DistinguishedName == "" {
-		return nil, fmt.Errorf("rtctrlSetNh %s not found", dn)
+		return nil, fmt.Errorf("rtctrl SetNh %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:292]:
```suggestion
		return nil, fmt.Errorf("Route Control Set Nexthop %s not found", dn)
```

File: aci/resource_aci_vmmvswitchpolicycont.go [hunk 9]
```diff
@@ -107,7 +107,7 @@ func getRemoteVSwitchPolicyGroup(client *client.Client, dn string) (*models.VSwi
 	}
 	vmmVSwitchPolicyCont := models.VSwitchPolicyGroupFromContainer(vmmVSwitchPolicyContCont)
 	if vmmVSwitchPolicyCont.DistinguishedName == "" {
-		return nil, fmt.Errorf("VSwitchPolicyGroup %s not found", vmmVSwitchPolicyCont.DistinguishedName)
+		return nil, fmt.Errorf("V Switch Policy Group %s not found", dn)
```
Comment (lhercot) [inline aci/resource_aci_vmmvswitchpolicycont.go:110]:
```suggestion
		return nil, fmt.Errorf("vSwitch Policy Group %s not found", dn)
```

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_aaalogindomain.go [hunk 10]
```diff
@@ -255,8 +254,7 @@ func resourceAciLoginDomainRead(ctx context.Context, d *schema.ResourceData, m i
 
 	aaaDomainAuth, err := getRemoteAuthenticationMethodfortheDomain(aciClient, childDn)
 	if err != nil {
-		d.SetId("")
-		return nil
+		return errorForObjectNotFound(err, dn, d)
```
[the remark this answers is not in the document]
Comment (shrsr) [reply aci/resource_aci_aaalogindomain.go:257]: Not required

File: aci/resource_aci_mgmtinboobzone.go [hunk 11]
```diff
@@ -154,7 +154,7 @@ func getRemoteOOBManagedNodesZone(client *client.Client, dn string) (*models.OOB
 	}
 	mgmtOobZone := models.OOBManagedNodesZoneFromContainer(mgmtOobZoneCont)
 	if mgmtOobZone.DistinguishedName == "" {
-		return nil, fmt.Errorf("OoB Managed Nodes Zone %s not found", dn)
+		return nil, fmt.Errorf("OOB Managed Nodes Zone %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_mgmtinboobzone.go:157]: if we use inband should we also use full description and use Out Of Band?

File: aci/resource_aci_rtctrlattrp.go [hunk 12]
```diff
@@ -359,7 +359,7 @@ func getRemoteRtctrlSetRedistMultipath(client *client.Client, dn string) (*model
 	}
 	rtctrlSetRedistMultipath := models.RedistributeMultipathActionFromContainer(rtctrlSetRedistMultipathCont)
 	if rtctrlSetRedistMultipath.DistinguishedName == "" {
-		return nil, fmt.Errorf("rtctrl Set Redist Multipath %s not found", dn)
+		return nil, fmt.Errorf("Route Control Set Redist Multipath %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_rtctrlattrp.go:362]: Should we change Redist into Redistribute?

File: aci/resource_aci_rtctrlattrp.go [hunk 13]
```diff
@@ -421,7 +421,7 @@ func getRemoteRtctrlSetDamp(client *client.Client, dn string) (*models.RtctrlSet
 	}
 	rtctrlSetDamp := models.RtctrlSetDampFromContainer(rtctrlSetDampCont)
 	if rtctrlSetDamp.DistinguishedName == "" {
-		return nil, fmt.Errorf("rtctrl Set Damp %s not found", dn)
+		return nil, fmt.Errorf("Route Control Set Damp %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_rtctrlattrp.go:424]: Should we change Damp into Dampening?

File: aci/resource_aci_mgmtinboobzone.go (hunk 11 above)
Comment (shrsr) [reply aci/resource_aci_mgmtinboobzone.go:157]: Ok

File: aci/resource_aci_rtctrlattrp.go (hunk 12 above)
Comment (shrsr) [reply aci/resource_aci_rtctrlattrp.go:362]: Sure

File: aci/resource_aci_rtctrlattrp.go (hunk 13 above)
Comment (shrsr) [reply aci/resource_aci_rtctrlattrp.go:424]: Yes

File: aci/resource_aci_bgpprotp.go [hunk 14]
```diff
@@ -65,7 +65,7 @@ func getRemoteL3outBGPProtocolProfile(client *client.Client, dn string) (*models
 	bgpProtP := models.L3outBGPProtocolProfileFromContainer(bgpProtPCont)
 
 	if bgpProtP.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outBGPProtocolProfile %s not found", bgpProtP.DistinguishedName)
+		return nil, fmt.Errorf("L3out BGP Protocol Profile %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_bgpprotp.go:68]: L3out -> L3Out

File: aci/resource_aci_bgprtsummpol.go [hunk 15]
```diff
@@ -70,7 +70,7 @@ func getRemoteBgpRouteSummarization(client *client.Client, dn string) (*models.B
 	bgpRtSummPol := models.BgpRouteSummarizationFromContainer(bgpRtSummPolCont)
 
 	if bgpRtSummPol.DistinguishedName == "" {
-		return nil, fmt.Errorf("BgpRouteSummarization %s not found", bgpRtSummPol.DistinguishedName)
+		return nil, fmt.Errorf("Bgp Route Summarization %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_bgprtsummpol.go:73]: Bgp -> BGP

File: aci/resource_aci_cloudepg.go [hunk 16]
```diff
@@ -157,7 +157,7 @@ func getRemoteCloudEPg(client *client.Client, dn string) (*models.CloudEPg, erro
 	cloudEPg := models.CloudEPgFromContainer(cloudEPgCont)
 
 	if cloudEPg.DistinguishedName == "" {
-		return nil, fmt.Errorf("CloudEPg %s not found", cloudEPg.DistinguishedName)
+		return nil, fmt.Errorf("Cloud EPg %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_cloudepg.go:160]: EPg -> EPG

File: aci/resource_aci_cloudextepg.go [hunk 17]
```diff
@@ -168,7 +168,7 @@ func getRemoteCloudExternalEPg(client *client.Client, dn string) (*models.CloudE
 	cloudExtEPg := models.CloudExternalEPgFromContainer(cloudExtEPgCont)
 
 	if cloudExtEPg.DistinguishedName == "" {
-		return nil, fmt.Errorf("CloudExternalEPg %s not found", cloudExtEPg.DistinguishedName)
+		return nil, fmt.Errorf("Cloud External EPg %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_cloudextepg.go:171]: EPg -> EPG

File: aci/resource_aci_cloudextepselector.go [hunk 18]
```diff
@@ -76,7 +76,7 @@ func getRemoteCloudEndpointSelectorforExternalEPgs(client *client.Client, dn str
 	cloudExtEPSelector := models.CloudEndpointSelectorforExternalEPgsFromContainer(cloudExtEPSelectorCont)
 
 	if cloudExtEPSelector.DistinguishedName == "" {
-		return nil, fmt.Errorf("CloudEndpointSelectorforExternalEPgs %s not found", cloudExtEPSelector.DistinguishedName)
+		return nil, fmt.Errorf("Cloud Endpoint Selector for External EPgs %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_cloudextepselector.go:79]: EPgs -> EPGs

File: aci/resource_aci_cloudrouterp.go [hunk 19]
```diff
@@ -93,7 +93,7 @@ func getRemoteCloudVpnGateway(client *client.Client, dn string) (*models.CloudVp
 	cloudRouterP := models.CloudVpnGatewayFromContainer(cloudRouterPCont)
 
 	if cloudRouterP.DistinguishedName == "" {
-		return nil, fmt.Errorf("CloudVpnGateway %s not found", cloudRouterP.DistinguishedName)
+		return nil, fmt.Errorf("Cloud Vpn Gateway %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_cloudrouterp.go:96]: Vpn -> VPN

File: aci/resource_aci_fvepgselector.go [hunk 20]
```diff
@@ -50,7 +50,7 @@ func getRemoteEndpointSecurityGroupEPgSelector(client *client.Client, dn string)
 	}
 	fvEPgSelector := models.EndpointSecurityGroupEPgSelectorFromContainer(fvEPgSelectorCont)
 	if fvEPgSelector.DistinguishedName == "" {
-		return nil, fmt.Errorf("EndpointSecurityGroupEPgSelector %s not found", fvEPgSelector.DistinguishedName)
+		return nil, fmt.Errorf("Endpoint Security Group EPg Selector %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_fvepgselector.go:53]: EPg -> EPG

File: aci/resource_aci_fvsubnet.go [hunk 21]
```diff
@@ -187,7 +187,7 @@ func getRemoteNlbEndpoint(client *client.Client, dn string) (*models.NlbEndpoint
 	}
 	fvEpNlb := models.NlbEndpointFromContainer(fvEpNlbCont)
 	if fvEpNlb.DistinguishedName == "" {
-		return nil, fmt.Errorf("NlbEndpoint %s not found", dn)
+		return nil, fmt.Errorf("Nlb Endpoint %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_fvsubnet.go:190]: Nlb -> NLB

File: aci/resource_aci_hsrpifp.go [hunk 22]
```diff
@@ -65,7 +65,7 @@ func getRemoteL3outHSRPInterfaceProfile(client *client.Client, dn string) (*mode
 	hsrpIfP := models.L3outHSRPInterfaceProfileFromContainer(hsrpIfPCont)
 
 	if hsrpIfP.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outHSRPInterfaceProfile %s not found", hsrpIfP.DistinguishedName)
+		return nil, fmt.Errorf("L3out HSRP Interface Profile %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_hsrpifp.go:68]: L3out -> L3Out

File: aci/resource_aci_hsrpsecvip.go [hunk 23]
```diff
@@ -72,7 +72,7 @@ func getRemoteL3outHSRPSecondaryVIP(client *client.Client, dn string) (*models.L
 	hsrpSecVip := models.L3outHSRPSecondaryVIPFromContainer(hsrpSecVipCont)
 
 	if hsrpSecVip.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outHSRPSecondaryVIP %s not found", hsrpSecVip.DistinguishedName)
+		return nil, fmt.Errorf("L3out HSRP Secondary VIP %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_hsrpsecvip.go:75]: L3out -> L3Out

File: aci/resource_aci_infrafexbndlgrp.go [hunk 24]
```diff
@@ -67,7 +67,7 @@ func getRemoteFexBundleGroup(client *client.Client, dn string) (*models.FexBundl
 	infraFexBndlGrp := models.FexBundleGroupFromContainer(infraFexBndlGrpCont)
 
 	if infraFexBndlGrp.DistinguishedName == "" {
-		return nil, fmt.Errorf("FexBundleGroup %s not found", infraFexBndlGrp.DistinguishedName)
+		return nil, fmt.Errorf("Fex Bundle Group %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_infrafexbndlgrp.go:70]: Fex -> FEX

File: aci/resource_aci_infraprovacc.go [hunk 25]
```diff
@@ -48,7 +48,7 @@ func getRemoteVlanEncapsulationforVxlanTraffic(client *client.Client, dn string)
 	infraProvAcc := models.VlanEncapsulationforVxlanTrafficFromContainer(infraProvAccCont)
 
 	if infraProvAcc.DistinguishedName == "" {
-		return nil, fmt.Errorf("VlanEncapsulationforVxlanTraffic %s not found", infraProvAcc.DistinguishedName)
+		return nil, fmt.Errorf("Vlan Encapsulation for Vxlan Traffic %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_infraprovacc.go:51]: Vlan -> VLAN

File: aci/resource_aci_ipnexthopp.go [hunk 26]
```diff
@@ -82,7 +82,7 @@ func getRemoteL3outStaticRouteNextHop(client *client.Client, dn string) (*models
 	ipNexthopP := models.L3outStaticRouteNextHopFromContainer(ipNexthopPCont)
 
 	if ipNexthopP.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outStaticRouteNextHop %s not found", ipNexthopP.DistinguishedName)
+		return nil, fmt.Errorf("L3out Static Route Next Hop %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_ipnexthopp.go:85]: L3out -> L3Out

File: aci/resource_aci_iproutep.go [hunk 27]
```diff
@@ -87,7 +87,7 @@ func getRemoteL3outStaticRoute(client *client.Client, dn string) (*models.L3outS
 	ipRouteP := models.L3outStaticRouteFromContainer(ipRoutePCont)
 
 	if ipRouteP.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outStaticRoute %s not found", ipRouteP.DistinguishedName)
+		return nil, fmt.Errorf("L3out Static Route %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_iproutep.go:90]: L3out -> L3Out

File: aci/resource_aci_l2extinstp.go [hunk 28]
```diff
@@ -186,7 +186,7 @@ func getRemoteL2outExternalEpg(client *client.Client, dn string) (*models.L2outE
 	l2extInstP := models.L2outExternalEpgFromContainer(l2extInstPCont)
 
 	if l2extInstP.DistinguishedName == "" {
-		return nil, fmt.Errorf("L2outExternalEpg %s not found", l2extInstP.DistinguishedName)
+		return nil, fmt.Errorf("L2out External Epg %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_l2extinstp.go:189]: L2out -> L2Out

File: aci/resource_aci_l3extmember.go [hunk 29]
```diff
@@ -81,7 +81,7 @@ func getRemoteL3outVPCMember(client *client.Client, dn string) (*models.L3outVPC
 	l3extMember := models.L3outVPCMemberFromContainer(l3extMemberCont)
 
 	if l3extMember.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outVPCMember %s not found", l3extMember.DistinguishedName)
+		return nil, fmt.Errorf("L3out VPC Member %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_l3extmember.go:84]: L3out -> L3Out

File: aci/resource_aci_l3extroutetagpol.go [hunk 30]
```diff
@@ -60,7 +60,7 @@ func getRemoteL3outRouteTagPolicy(client *client.Client, dn string) (*models.L3o
 	l3extRouteTagPol := models.L3outRouteTagPolicyFromContainer(l3extRouteTagPolCont)
 
 	if l3extRouteTagPol.DistinguishedName == "" {
-		return nil, fmt.Errorf("L3outRouteTagPolicy %s not found", l3extRouteTagPol.DistinguishedName)
+		return nil, fmt.Errorf("L3out Route Tag Policy %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_l3extroutetagpol.go:63]: L3out -> L3Out

File: aci/resource_aci_mgmtinboobzone.go [hunk 31]
```diff
@@ -142,7 +142,7 @@ func getRemoteInBManagedNodesZone(client *client.Client, dn string) (*models.InB
 	}
 	mgmtInBZone := models.InBManagedNodesZoneFromContainer(mgmtInBZoneCont)
 	if mgmtInBZone.DistinguishedName == "" {
-		return nil, fmt.Errorf("InBManagedNodesZone %s not found", mgmtInBZone.DistinguishedName)
+		return nil, fmt.Errorf("InBand Managed Nodes Zone %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_mgmtinboobzone.go:145]: InBand -> In-Band

File: aci/resource_aci_ospfrtsummpol.go [hunk 32]
```diff
@@ -77,7 +77,7 @@ func getRemoteOspfRouteSummarization(client *client.Client, dn string) (*models.
 	ospfRtSummPol := models.OspfRouteSummarizationFromContainer(ospfRtSummPolCont)
 
 	if ospfRtSummPol.DistinguishedName == "" {
-		return nil, fmt.Errorf("OspfRouteSummarization %s not found", ospfRtSummPol.DistinguishedName)
+		return nil, fmt.Errorf("Ospf Route Summarization %s not found", dn)
```
Comment (anvitha-jain) [inline aci/resource_aci_ospfrtsummpol.go:80]: Ospf -> OSPF

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

**Review by anvitha-jain — DISMISSED** [review]
LGTM

File: aci/resource_aci_aaaldapgroupmapruleref.go [hunk 33]
```diff
@@ -45,7 +45,7 @@ func getRemoteLDAPGroupMapruleref(client *client.Client, dn string) (*models.LDA
 	}
 	aaaLdapGroupMapRuleRef := models.LDAPGroupMaprulerefFromContainer(aaaLdapGroupMapRuleRefCont)
 	if aaaLdapGroupMapRuleRef.DistinguishedName == "" {
-		return nil, fmt.Errorf("LDAPGroupMapruleref %s not found", aaaLdapGroupMapRuleRef.DistinguishedName)
+		return nil, fmt.Errorf("LDAP Group Map rule ref %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_aaaldapgroupmapruleref.go:48]: should we capitalise Rule Ref? In aci/resource_aci_aaaldapgroupmaprule.go you do use Rule capitalised.

File: aci/resource_aci_bgpbestpathctrlpol.go [hunk 34]
```diff
@@ -64,7 +64,7 @@ func getRemoteBgpBestPathPolicy(client *client.Client, dn string) (*models.BgpBe
 	bgpBestPathCtrlPol := models.BgpBestPathPolicyFromContainer(bgpBestPathCtrlPolCont)
 
 	if bgpBestPathCtrlPol.DistinguishedName == "" {
-		return nil, fmt.Errorf("BgpBestPathPolicy %s not found", bgpBestPathCtrlPol.DistinguishedName)
+		return nil, fmt.Errorf("Bgp Best Path Policy %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_bgpbestpathctrlpol.go:67]: Should we use BGP since it is a abbreviation?

File: aci/resource_aci_cloudapp.go [hunk 35]
```diff
@@ -54,7 +54,7 @@ func getRemoteCloudApplicationcontainer(client *client.Client, dn string) (*mode
 	cloudApp := models.CloudApplicationcontainerFromContainer(cloudAppCont)
 
 	if cloudApp.DistinguishedName == "" {
-		return nil, fmt.Errorf("CloudApplicationcontainer %s not found", cloudApp.DistinguishedName)
+		return nil, fmt.Errorf("Cloud Application container %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_cloudapp.go:57]: Should we capitalise container?

File: aci/resource_aci_cloudextepg.go (hunk 17 above)
Comment (akinross) [reply aci/resource_aci_cloudextepg.go:171]: See this is marked as resolved but is still EPg? See other usage of EPG so should this not be changed?

File: aci/resource_aci_fvbd.go [hunk 36]
```diff
@@ -313,7 +313,7 @@ func getRemoteBridgeDomain(client *client.Client, dn string) (*models.BridgeDoma
 	fvBD := models.BridgeDomainFromContainer(fvBDCont)
 
 	if fvBD.DistinguishedName == "" {
-		return nil, fmt.Errorf("BridgeDomain %s not found", fvBD.DistinguishedName)
+		return nil, fmt.Errorf("Bridge Domain %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_fvbd.go:316]: I saw earlier mentioning of BD, should we use abbreviated forms or full form? This is generic question that applies for all other object mentions. @lhercot any thoughts?

File: aci/resource_aci_fvrscloudaccount.go [hunk 37]
```diff
@@ -45,7 +45,7 @@ func getRemoteTenantToCloudAccountAssociation(client *client.Client, dn string)
 	}
 	fvRsCloudAccount := models.TenantToCloudAccountAssociationFromContainer(fvRsCloudAccountCont)
 	if fvRsCloudAccount.DistinguishedName == "" {
-		return nil, fmt.Errorf("Tenant To Cloud Account Association %s not found", fvRsCloudAccount.DistinguishedName)
+		return nil, fmt.Errorf("Tenant To Cloud Account Association %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_fvrscloudaccount.go:48]: Should be use lower case for To and Association?

File: aci/resource_aci_infraaccbndlgrp.go [hunk 38]
```diff
@@ -201,7 +201,7 @@ func getRemotePCVPCInterfacePolicyGroup(client *client.Client, dn string) (*mode
 	infraAccBndlGrp := models.PCVPCInterfacePolicyGroupFromContainer(infraAccBndlGrpCont)
 
 	if infraAccBndlGrp.DistinguishedName == "" {
-		return nil, fmt.Errorf("PCVPCInterfacePolicyGroup %s not found", infraAccBndlGrp.DistinguishedName)
+		return nil, fmt.Errorf("PCVPC Interface Policy Group %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_infraaccbndlgrp.go:204]: should we use PC/VPC since these are seperated?

File: aci/resource_aci_infraaccbndlsubgrp.go [hunk 39]
```diff
@@ -51,7 +51,7 @@ func getRemoteOverridePCVPCPolicyGroup(client *client.Client, dn string) (*model
 	}
 	infraAccBndlSubgrp := models.OverridePCVPCPolicyGroupFromContainer(infraAccBndlSubgrpCont)
 	if infraAccBndlSubgrp.DistinguishedName == "" {
-		return nil, fmt.Errorf("OverridePCVPCPolicyGroup %s not found", dn)
+		return nil, fmt.Errorf("Override PCVPC Policy Group %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_infraaccbndlsubgrp.go:54]: should we use PC/VPC since these are seperated?

File: aci/resource_aci_infrarsdomp.go [hunk 40]
```diff
@@ -50,7 +50,7 @@ func getRemoteInfraRsDomP(client *client.Client, dn string) (*models.InfraRsDomP
 	}
 	infraRsDomP := models.InfraRsDomPFromContainer(infraRsDomPCont)
 	if infraRsDomP.DistinguishedName == "" {
-		return nil, fmt.Errorf("InfraRsDomP %s not found", dn)
+		return nil, fmt.Errorf("Infra RsDomP %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_infrarsdomp.go:53]: Should we reword RsDomP to something like Domain?

File: aci/resource_aci_l3extloopbackifp.go [hunk 41]
```diff
@@ -55,7 +55,7 @@ func getRemoteLoopBackInterfaceProfile(client *client.Client, dn string) (*model
 	l3extLoopBackIfP := models.LoopBackInterfaceProfileFromContainer(l3extLoopBackIfPCont)
 
 	if l3extLoopBackIfP.DistinguishedName == "" {
-		return nil, fmt.Errorf("LoopBackInterfaceProfile %s not found", l3extLoopBackIfP.DistinguishedName)
+		return nil, fmt.Errorf("LoopBack Interface Profile %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_l3extloopbackifp.go:58]: Should we use Loop Back Interface Profile?

File: aci/resource_aci_mgmtconnectivityprefs.go [hunk 42]
```diff
@@ -46,7 +46,7 @@ func getRemoteMgmtconnectivitypreference(client *client.Client, dn string) (*mod
 	}
 	mgmtConnectivityPrefs := models.MgmtconnectivitypreferenceFromContainer(mgmtConnectivityPrefsCont)
 	if mgmtConnectivityPrefs.DistinguishedName == "" {
-		return nil, fmt.Errorf("Mgmtconnectivitypreference %s not found", mgmtConnectivityPrefs.DistinguishedName)
+		return nil, fmt.Errorf("Mgmt connectivity preference %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_mgmtconnectivityprefs.go:49]: Should we capitalise?

File: aci/resource_aci_mgmtrsoobstnode.go [hunk 43]
```diff
@@ -100,7 +100,7 @@ func getRemoteOutofbandStaticNode(client *client.Client, dn string) (*models.Out
 	mgmtRsOoBStNode := models.OutofbandStaticNodeFromContainer(mgmtRsOoBStNodeCont)
 
 	if mgmtRsOoBStNode.DistinguishedName == "" {
-		return nil, fmt.Errorf("Out of Band Static Node %s not found", mgmtRsOoBStNode.DistinguishedName)
+		return nil, fmt.Errorf("Out-of-Band Static Node %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_mgmtrsoobstnode.go:103]: Should we use  -Of- for consistency?

File: aci/resource_aci_mgmtinboobzone.go [hunk 44]
```diff
@@ -154,7 +154,7 @@ func getRemoteOOBManagedNodesZone(client *client.Client, dn string) (*models.OOB
 	}
 	mgmtOobZone := models.OOBManagedNodesZoneFromContainer(mgmtOobZoneCont)
 	if mgmtOobZone.DistinguishedName == "" {
-		return nil, fmt.Errorf("InBManagedNodesZone %s not found", mgmtOobZone.DistinguishedName)
+		return nil, fmt.Errorf("Out-Of-Band Managed Nodes Zone %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_mgmtinboobzone.go:157]: Should we use -Of- for consistency below or both use -of-?

File: aci/resource_aci_nodemgmtepg.go [hunk 45]
```diff
@@ -286,7 +286,7 @@ func getRemoteOutOfBandManagementEPg(client *client.Client, dn string) (*models.
 	mgmtOoB := models.OutOfBandManagementEPgFromContainer(mgmtOoBCont)
 
 	if mgmtOoB.DistinguishedName == "" {
-		return nil, fmt.Errorf("OutOfBandManagementEPg %s not found", mgmtOoB.DistinguishedName)
+		return nil, fmt.Errorf("Out-Of-Band Management EPG %s not found", dn)
```
Comment (akinross) [inline aci/resource_aci_nodemgmtepg.go:289]: Should we use -Of- for consistency or -of-?

File: aci/resource_aci_infrarsdomp.go (hunk 40 above)
Comment (shrsr) [reply aci/resource_aci_infrarsdomp.go:53]: I looked it up and couldn't find any logical term for it. Did you mean RsDomainP?

File: aci/resource_aci_fvbd.go (hunk 36 above)
Comment (shrsr) [reply aci/resource_aci_fvbd.go:316]: Leaving it as Bridge Domain, seeing that the name of the function is the same

**Review by anvitha-jain — APPROVED** [review]

File: aci/resource_aci_l3extloopbackifp.go [hunk 46]
```diff
@@ -55,7 +55,7 @@ func getRemoteLoopBackInterfaceProfile(client *client.Client, dn string) (*model
 	l3extLoopBackIfP := models.LoopBackInterfaceProfileFromContainer(l3extLoopBackIfPCont)
 
 	if l3extLoopBackIfP.DistinguishedName == "" {
-		return nil, fmt.Errorf("LoopBackInterfaceProfile %s not found", l3extLoopBackIfP.DistinguishedName)
+		return nil, fmt.Errorf("Loop Back Interface Profile %s not found", dn)
```
[the remark this answers is not in the document]
Comment (lhercot) [reply aci/resource_aci_l3extloopbackifp.go:58]: Not fixed

File: aci/resource_aci_infrarsdomp.go (hunk 40 above)
Comment (lhercot) [reply aci/resource_aci_infrarsdomp.go:53]: Relationship to Domain Profile

File: aci/resource_aci_l3extloopbackifp.go (hunk 46 above)
[the remark this answers is not in the document]
Comment (shrsr) [reply aci/resource_aci_l3extloopbackifp.go]: done

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**